File: Java/reverserot.java

// https://open.kattis.com/problems/reverserot
import java.util.Scanner;

public class reverserot {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String alpha = "ABCDEFGHIJKLMNOPQRSTUVWXYZ_.";

        while (true) {
            int rot = sc.nextInt();
            if (rot == 0)
                break;
            char[] arr = sc.nextLine().substring(1).toCharArray();
            String decode = "";
            for (int a = arr.length - 1; a >= 0; a--) {
                int index = 0;
                for (int x = 0; x < alpha.length(); x++)
                    if (alpha.charAt(x) == arr[a]) {
                        index = x + rot;
                        break;
                    }
                index %= alpha.length();
                decode += alpha.charAt(index);
            }
            System.out.println(decode);
        }
        sc.close();
    }
}

File: C#/reverserot.cs
using System;

public class reverserot {
    public static void Main (string[] args) {
        string alpha = "ABCDEFGHIJKLMNOPQRSTUVWXYZ_.";
        for (;;) {
            string[] line = Console.ReadLine ().Split (' ');
            int rot = int.Parse (line[0]);
            if (rot == 0)
                break;
            string msg = line[1];
            string s = "";
            for (int x = msg.Length - 1; x >= 0; x--)
                for (int a = 0; a < alpha.Length; a++)
                    if (alpha[a] == msg[x]) {
                        s += alpha[(a + rot) % alpha.Length];
                        break;
                    }
            Console.WriteLine (s);
        }
    }
}